langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/TestHelper.java

```diff
@@ -0,0 +1,116 @@
+
+    private final String dbName;
+    private final MongoClient client;
+    private MongoDbEmbeddingStore embeddingStore;
+
+    TestHelper(MongoClient client) {
+        this.dbName = "test_database_" + ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
+        this.client = client;
+    }
+
+    TestHelper initialize() {
+        return this.initialize(b -> b);
+    }
+
+    TestHelper initialize(Function<MongoDbEmbeddingStore.Builder, MongoDbEmbeddingStore.Builder> initializer) {
+        this.embeddingStore = initializer.apply(getDefaultMongoDbEmbeddingStoreBuilder()).build();
+        return this;
+    }
+
+    static void assertDoContainerTests() {
+        Assumptions.assumeTrue(getenv("MONGODB_DISABLE_CONTAINER_TESTS") == null);
+    }
+
+    EmbeddingStore<TextSegment> getEmbeddingStore() {
+        return embeddingStore;
+    }
+
+    static MongoClient createClientFromContainer(MongoDBAtlasLocalContainer mongodb) {
+        MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(mongodb.getConnectionString()));
+        return createClientFromBuilder(builder);
+    }
+
+    static MongoClient createClientFromEnv() {
+        boolean hasParts = HOST != null && USERNAME != null && PASSWORD != null;
+        Assumptions.assumeTrue(hasParts || CONNECTION_STRING != null,
+                "Skipping test: host/username/password, or connection string must be provided via env vars.");
+
+        String connectionString = CONNECTION_STRING != null
+                ? CONNECTION_STRING
+                : format("mongodb+srv://%s:%s@%s/?retryWrites=true&w=majority", USERNAME, PASSWORD, HOST);
+        MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(connectionString));
+        return createClientFromBuilder(builder);
+    }
+
+    private static MongoClient createClientFromBuilder(MongoClientSettings.Builder builder) {
+        MongoClientSettings mongoClientSettings = builder
+                .uuidRepresentation(UuidRepresentation.STANDARD)
+                .serverApi(ServerApi.builder().version(ServerApiVersion.V1).build())
+                .build();
+        return MongoClients.create(mongoClientSettings);
+    }
+
+    private MongoDbEmbeddingStore.Builder getDefaultMongoDbEmbeddingStoreBuilder() {
+        IndexMapping indexMapping = IndexMapping.builder()
+                .dimension(EMBEDDING_MODEL.dimension())
+                .metadataFieldNames(Sets.newHashSet("test-key"))
+                .build();
+        return MongoDbEmbeddingStore.builder()
+                .fromClient(client)
+                .databaseName(dbName)
+                .collectionName("test_collection")
+                .indexName("test_index")
+                .indexMapping(indexMapping)
+                .createIndex(true);
+    }
+
+    void afterTests() {
+        getDatabase().drop();
```

Comment:
Looks like `MongoDB Atlas Cloud` does not allow user to drop database. 

```java
com.mongodb.MongoCommandException: Command failed with error 8000 (AtlasError): 'user is not allowed to do action [dropDatabase] on [test_database_1062281260.]' on server ac-fte18bh-shard-00-02.eo9fifu.mongodb.net:27017. The full response is {"ok": 0, "errmsg": "user is not allowed to do action [dropDatabase] on [test_database_1062281260.]", "code": 8000, "codeName": "AtlasError"}

	at com.mongodb.internal.connection.ProtocolHelper.getCommandFailureException(ProtocolHelper.java:210)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:520)
	at com.mongodb.internal.connection.InternalStreamConnection.sendAndReceiveInternal(InternalStreamConnection.java:448)
	at com.mongodb.internal.connection.InternalStreamConnection.lambda$sendAndReceive$0(InternalStreamConnection.java:375)
	at com.mongodb.internal.connection.InternalStreamConnection.sendAndReceive(InternalStreamConnection.java:378)
	at com.mongodb.internal.connection.UsageTrackingInternalConnection.sendAndReceive(UsageTrackingInternalConnection.java:111)
	at com.mongodb.internal.connection.DefaultConnectionPool$PooledConnection.sendAndReceive(DefaultConnectionPool.java:748)
	at com.mongodb.internal.connection.CommandProtocolImpl.execute(CommandProtocolImpl.java:68)
	at com.mongodb.internal.connection.DefaultServer$DefaultServerProtocolExecutor.execute(DefaultServer.java:208)
	at com.mongodb.internal.connection.DefaultServerConnection.executeProtocol(DefaultServerConnection.java:113)
	at com.mongodb.internal.connection.DefaultServerConnection.command(DefaultServerConnection.java:82)
	at com.mongodb.internal.connection.DefaultServerConnection.command(DefaultServerConnection.java:73)
	at com.mongodb.internal.connection.DefaultServer$OperationCountTrackingConnection.command(DefaultServer.java:298)
	at com.mongodb.internal.operation.SyncOperationHelper.executeCommand(SyncOperationHelper.java:231)
	at com.mongodb.internal.operation.DropDatabaseOperation.lambda$execute$0(DropDatabaseOperation.java:61)
	at com.mongodb.internal.operation.SyncOperationHelper.withConnectionSource(SyncOperationHelper.java:167)
	at com.mongodb.internal.operation.SyncOperationHelper.withConnection(SyncOperationHelper.java:112)
	at com.mongodb.internal.operation.DropDatabaseOperation.execute(DropDatabaseOperation.java:60)
	at com.mongodb.internal.operation.DropDatabaseOperation.execute(DropDatabaseOperation.java:45)
	at com.mongodb.client.internal.MongoClusterImpl$OperationExecutorImpl.execute(MongoClusterImpl.java:379)
	at com.mongodb.client.internal.MongoDatabaseImpl.executeDrop(MongoDatabaseImpl.java:230)
	at com.mongodb.client.internal.MongoDatabaseImpl.drop(MongoDatabaseImpl.java:220)
	at dev.langchain4j.store.embedding.mongodb.TestHelper.afterTests(TestHelper.java:101)
	at dev.langchain4j.store.embedding.mongodb.MongoDbEmbeddingStoreLegacyIndexesIT.afterEach(MongoDbEmbeddingStoreLegacyIndexesIT.java:50)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

Reply:
There are a couple of related issues here: whether the databases should be dropped during tests, and the implications for use of M0 Atlas for testing.

I used a new database with a random name for each test in this PR. The indexes are eventually consistent, so it is possible for tests to fail in confusing ways. This seemed easier to debug and less error-prone, and it was not slower. I believe that if it is possible to test in this way, it is preferable.

It should be possible to test against only the `MongoDBAtlasLocalContainer`. This (along with the local Atlas CLI) are the recommended ways to test. I did not want to remove the M0 Atlas tests in this PR without discussing first, but I propose testing exclusively with the container. Though, if there is some reason not to do this, that is fine, and I would look into how to address that. Let me know!

Reply:
1. For local container, it doesn't matter right? For cloud, we reuse the same database and collection at any test time, and we will delete all datas before each test. And just keep the database and collection on cloud.
2. Tests in `Atlas Cloud` is just for consistence. We don't want to see that users failed to embed in the cloud. (Because we don't know the difference between cloud and local, such as cloud do not allow users to drop database using command).

Reply:
1. The search indexes are still eventually consistent on a local container. The index itself is technically data - to give an example similar to something I ran into during this PR, consider an index that has been marked for deletion still being present when the next test runs and attempts to use/create that index. Dropping will ensure that each test starts with a clean slate.

2. I believe there is no difference between cloud and local. The drop database problem above appears to be a [permissions issue](https://stackoverflow.com/questions/55636442), and the index creation limitation was removed in August (this [page](https://www.mongodb.com/docs/atlas/atlas-search/shared-tier-limitations/) discusses limitations, and will be updated soon to reflect this).

If there is no longer an objection to removing cloud tests (2), then I think this would resolve the problem. Otherwise, perhaps we could try updating the permissions for the test account?

Reply:
Sounds reasonable, maybe we can discard Cloud test after `MongoDB Atlas` update the drop database permission. And now could we just keep it?

Reply:
Apologies, I should clarify - the drop database permission is something that users who plan to test against cloud would need to have configured on their Atlas account. The default account has this permission, but if you create secondary accounts, those will not by default. However, to avoid this matter entirely, I have changed the tests to use a random collection name (instead of a database) in this [PR](https://github.com/langchain4j/langchain4j/pull/1932), and rebased. I believe this removes any impediment to removing the cloud test?

I have set tests up to check if connection string or host/user/pass env vars are set. If they are not set, the test will use local container testing. The env vars are present now in case devs want to use Atlas CLI, and save a few seconds on test startup. I have removed the env vars from the workflow, and this should disable cloud testing. I have also checked with other developers and there should be no issue with testing against the container, especially after the recent upgrade to the MongoDB testcontainer.

I ran the PR's tests against a) the local container, and b) a cloud instance using an account that had write but not admin permissions, and they passed.